File: 3550-smallest-index-with-digit-sum-equal-to-index/solution.py
class Solution:
    def smallestIndex(self, nums: list[int]) -> int:
        def get_digit_sum(n: int) -> int:
            total = 0
            while n > 0:
                total += n % 10
                n //= 10
            return total

        for i, num in enumerate(nums):
            # Check if the sum of digits equals the current index
            if get_digit_sum(num) == i:
                return i  # Return immediately as enumerate iterates in increasing order

        return -1  # No valid index found

File: 3550-smallest-index-with-digit-sum-equal-to-index/solution.cpp
#include <vector>

class Solution {
private:
    // Helper function to calculate the sum of digits of a non-negative integer
    int getDigitSum(int n) {
        int sum = 0;
        while (n > 0) {
            sum += n % 10;
            n /= 10;
        }
        return sum;
    }

public:
    int smallestIndex(std::vector<int>& nums) {
        for (int i = 0; i < static_cast<int>(nums.size()); ++i) {
            // Check if the sum of digits equals the current index
            if (getDigitSum(nums[i]) == i) {
                return i; // Return immediately as we iterate in increasing order
            }
        }
        return -1; // No valid index found
    }
};